src/DataProtection/DataProtection/src/KeyManagement/KeyRingBasedDataProtector.cs

```diff
@@ -381,6 +353,118 @@ public byte[] GetAadForKey(Guid keyId, bool isProtecting)
             }
         }
 
+#if NET10_0_OR_GREATER
+        public AdditionalAuthenticatedDataTemplate(string[] purposes)
```

Comment:
Because we're replacing an existing implementation, I'd like to see at least one known payload correctness test, i.e.

- run the *old* code on something like an array of 5 strings, ideally with at least one of those strings being more than 127 utf8 bytes, and ideally containing some exotic unicode characters
- capture the return value from that, and get the base-64 (Convert.ToBase64String) or hex (BitConverter.ToString), so we can include it in a test trivially
- add a test that runs the new code for that some input, and compares the base-64 or hex output for equality

Reply:
added `AdditionalAuthenticatedDataTemplateBuildAadTemplateBytes_ReturnsSameResultAsPreviousImplementation()`